activerecord/test/cases/relations_test.rb

```diff
@@ -63,6 +63,16 @@ def test_multivalue_where
     assert_equal 1, posts.to_a.size
   end
 
+  def test_wharel
+    posts = Post.wharel { |post| (post.author_id != 1).and(post.id > 2) }
+    assert_equal 6, posts.to_a.size
+  end
+
+  def test_wharel_can_chain
+    posts = Post.wharel { |post| post.author_id >= 1 }.wharel { |post| post.id <= 4 }
```

Comment:
I do like this abstraction, however, chances are the `<=`, `>=` ... abstractions may be controversial. IMO I'd go without these abstractions, make it simple, and if the PR without it is merged, then you can suggest this one, WDYT?